Zend/zend_gc.h

```diff
@@ -46,6 +46,20 @@ ZEND_API bool gc_enabled(void);
 ZEND_API bool gc_protect(bool protect);
 ZEND_API bool gc_protected(void);
 
+/* Arrays are non-collectable by default and will be marked as collectable only
+ * when they can form a cycle. This is only the case then they contain objects
+ * or references. */
+static zend_always_inline bool gc_ht_check_collectable(HashTable *ht) {
+	return (GC_FLAGS(ht) & (GC_PERSISTENT|GC_NOT_COLLECTABLE)) == GC_NOT_COLLECTABLE;
+}
+static zend_always_inline bool gc_ht_value_may_cause_cycle(zval *zv) {
+	return (Z_TYPE_P(zv) & (IS_OBJECT|IS_REFERENCE)) != 0;
+}
```

Comment:
This conditions seems wrong. Z_TYPE_P() is not a bitset. This should be rewritten as ``Z_TYPE_P(zv) == IS_OBJECT || Z_TYPE_P(zv) == IS_REFERENCE``.

Reply:
Oh, ups. You're right of course. I changed this last minute.